pkg/kubectl/cmd/util/factory.go

```diff
@@ -274,12 +274,62 @@ func makeInterfacesFor(versionList []unversioned.GroupVersion) func(version unve
 	}
 }
 
+type DiscoveryExpander struct {
```

Comment:
Can you factor this to accept a `DiscoveryClient` and add it to the `ShortcutExpander` I've added in https://github.com/kubernetes/kubernetes/pull/32493?